test_scripts/tests_runner.py

```diff
@@ -0,0 +1,181 @@
+import os
+import sys
+import logging
+import pathlib
+import argparse
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[1]))
+from common.helper import cmd_exec
+from common.logger_conf import configure_logger
+from common.manifest_manager import Manifest
+
+
+class TestsRunnerException(Exception):
+    pass
+
+
+class ArtifactsNotFoundException(TestsRunnerException):
+    pass
+
+
+class TestScenarioNotFoundException(TestsRunnerException):
+    pass
+
+
+class Action(object):
+    """
+    Command line script runner
+    """
+
+    def __init__(self, name, cmd, work_dir, env, verbose):
+        """
+        :param name: Name of action
+        :type name: String
+
+        :param cmd: command line script
+        :type cmd: String
+
+        :param work_dir: Path where script will execute
+        :type work_dir: pathlib.Path
+
+        :param env: Environment variables for script
+        :type env: Dict
+
+        :param verbose: Flag for output all logs
+        :type verbose: Boolean
+        """
+
+        self.cmd = cmd
+        self.work_dir = work_dir
+        self.env = env
+        self.verbose = verbose
+
+        self.log = logging.getLogger(name)
+
+    def run(self):
+        """
+        Script runner
+
+        :return: None | subprocess.CalledProcessError
+        """
+
+        self.log.info('-' * 50)
+
+        if self.cmd:
+            if isinstance(self.cmd, list):
+                self.cmd = ' && '.join(self.cmd)
+
+            env = os.environ.copy()
+
+            if self.env:
+                env.update(self.env)
+
+            if self.work_dir:
+                self.work_dir.mkdir(parents=True, exist_ok=True)
+
+            error_code, out = cmd_exec(self.cmd, env=env, cwd=self.work_dir, log=self.log)
+
+            if error_code:
+                self.log.error(out)
+                self.log.error('failed')
+            else:
+                if self.verbose:
+                    self.log.info(out)
+                    self.log.info('completed')
+                else:
+                    self.log.debug(out)
+                    self.log.debug('completed')
+
+            return error_code
+
+
+class TestRunner:
```

Comment:
probably TestGenerator?
Probably can be base class for BuildGenerator and TestGenerator